core/src/test/java/kafka/server/SharePartitionManagerTest.java

```diff
@@ -92,8 +94,18 @@ public class SharePartitionManagerTest {
 
     private static final int PARTITION_MAX_BYTES = 40000;
     static final int RECORD_LOCK_DURATION_MS = 30000;
-    static final Timer TIMER = new SystemTimerReaper("sharePartitionTestReaper",
-            new SystemTimer("sharePartitionTestTimer"));
+    private static Timer mockTimer;
+
+    @BeforeEach
+    public void setUp() {
+        mockTimer = new SystemTimerReaper("sharePartitionTestReaper",
+                new SystemTimer("sharePartitionTestTimer"));
+    }
+
+    @AfterEach
+    public void tearDown() throws Exception {
+        mockTimer.close();
```

Comment:
Can we have a test which ensures that closing the SPM actually closes the timer?

Reply:
Hi Andrew, I've added the test `testCloseSharePartitionManager` to demonstrate the above functionality